BlueprintUI/Sources/Layout/UnitPoint.swift

```diff
@@ -0,0 +1,84 @@
+import CoreGraphics
+
+/// A normalized point in an element's coordinate space.
+///
+/// Use a unit point to represent a location in an element without having to know the element’s
+/// rendered size. The point stores a value in each dimension that indicates the fraction of the
+/// element’s size in that dimension — measured from the element’s origin — where the point appears.
+/// For example, you can create a unit point that represents the center of any element by using the
+/// value `0.5` for each dimension:
+///
+/// ```swift
+/// let unitPoint = UnitPoint(x: 0.5, y: 0.5)
+/// ```
+///
+/// To project the unit point into the rendered element’s coordinate space, multiply each component
+/// of the unit point with the corresponding component of the element’s size:
+///
+/// ```swift
+/// let projectedPoint = CGPoint(
+///     x: unitPoint.x * size.width,
+///     y: unitPoint.y * size.height
+/// )
+/// ```
+///
+/// You can perform this calculation yourself if you happen to know an element’s size, but Blueprint
+/// typically does this for you to carry out operations that you request, like when you place a
+/// subelement in a custom layout.
+///
+/// You can create custom unit points with explicit values, like the example above, or you can use
+/// one of the built-in unit points, like ``zero``, ``center``, or ``topTrailing``. The built-in
+/// values correspond to the alignment positions of the similarly named, built-in ``Alignment``
+/// types.
+///
+/// - Note: A unit point with one or more components outside the range `[0, 1]` projects to a point
+///   outside of the element.
+///
+public struct UnitPoint: Hashable {
```

Comment:
Maybe its just me, but I found this name somewhat confusing, perhaps `FractionPoint` or similar?

Reply:
I believe this is inspired by [SwiftUI's UnitPoint](https://developer.apple.com/documentation/swiftui/unitpoint).

Reply:
Yep, taken directly from SwiftUI. I assume the name has to do with the idea that this is a point on [the unit square](https://en.wikipedia.org/wiki/Unit_square).